src/librustc/mir/cache.rs

```diff
@@ -159,7 +147,14 @@ impl<'a, 'tcx> BorrowedCache<'a, 'tcx> {
     }
 }
 
-impl<'a, 'tcx> Index<BasicBlock> for BorrowedCache<'a, 'tcx> {
+impl<'a, 'tcx> Deref for BodyCache<&'a Body<'tcx>> {
```

Comment:
Is it alright to consider `BodyCache` as "smart pointer" for `Body`? The docs on `Deref` say it should only be implemented for smart pointers, and I wasn't sure if this counted or not. This significantly reduces the number of lines of code that have to be changed, but I'm not sure if this will cause a lot of confusion during debugging.

Reply:
yea, I think this is ok.